### ISSUE TITLE
Blocking candidates used only in Test set

### ISSUE THREAD

**Issue by Asciotti** [ePBRN/Medical-Record-Linkage-Ensemble#1]
Hi,

I'd like to preface all this with an appreciation for releasing code that actually runs. We are replicating this paper as part of a grad course and it's fantastic for code in published literature to _work_.

I was looking through your paper and the code and I'm confused about why the blocking candidates are only utilized in the test set. Relevant code snippets are below...

Within ePBRN replications [here](https://github.com/ePBRN/Medical-Record-Linkage-Ensemble/blob/master/ePBRN_UNSW_Linkage.ipynb), we see 
```
# Training data
df_train = pd.read_csv(trainset+".csv", index_col = "rec_id")
train_true_links = generate_true_links(df_train)
...
X_train, y_train = generate_train_X_y(df_train)
```

```
# Testing data
df_test = pd.read_csv(testset+".csv", index_col = "rec_id")
test_true_links = generate_true_links(df_test)
...
df_X_test = extract_features(df_test, all_candidate_pairs)
```

Under the hood of `generate_train_X_y` is `extract_features`, which does not use any of the blocking candidates. Why is this? The way I understand the situation is that determining and using blocking candidates is essentially a pre-filtering step. I am confused why it would show up in the test but not training - is it because the training data we "have" the truth so pre-filtering just reduces our training set? If you have any info on that, would greatly appreciate it. 

Also, this behavior is similar in the FEBRL schema experiments.
